### PR TITLE
liblcvm: extend get_frame_drop_info() to provide a configurable perce…

### PR DESCRIPTION
**Pull request by leela9980**
Extended get_frame_drop_info to support total frame drop count, frame_drop_length_percentile.

### CODE REVIEW

File: include/liblcvm.h [hunk 1]
```diff
@@ -21,13 +21,18 @@
...
-// @param[in] debug: Debug level.
+// @param[in] percentile: Percentile.
+// @param[out] frame_drop_length_percentile: Frame drop length percentile.
+// @param[in] debug: Debug level.    
 int get_frame_drop_info(const char *infile, int *num_video_frames,
-                        float *frame_rate_fps, float *frame_drop_ratio,
-                        float *normalized_frame_drop_average_length, int debug);
+                        float *frame_rate_fps, int *frame_drop_count,
+                        float *frame_drop_ratio, float *normalized_frame_drop_average_length,
+                        float percentile, float *frame_drop_length_percentile,
```
Comment (chemag) [inline include/liblcvm.h:34]: Line 34 should be:

                        const std::vector<float> percentile_list, std::vector<float> &frame_drop_length_percentile_list,

File: include/liblcvm.h [hunk 2]
```diff
@@ -21,13 +21,18 @@
 // @param[in] infile: Name of the file to be parsed.
 // @param[out] num_video_frames: Number of video frames in the file.
 // @param[out] frame_rate_fps: Frame rate (fps).
+// @param[out] frame_drop_count: Frame drop count.
```
Comment (chemag) [inline include/liblcvm.h:24]: please add this in a separate patch

File: src/liblcvm.cc [hunk 3]
```diff
@@ -212,17 +214,22 @@ int get_frame_drop_info(const char *infile, int *num_video_frames,
 
   // 6. calculate frame drop ratio as extra drop length over total duration
   *frame_drop_ratio = drop_length_duration_sec / total_duration_sec;
-
+  *frame_drop_count = drop_length_sec_list_sum / delta_timestamp_sec_median;
```
Comment (chemag) [inline src/liblcvm.cc:217]: This looks wrong. Why are we dividing to calculate the frame drop count?

File: src/liblcvm.cc [hunk 4]
```diff
@@ -212,17 +214,22 @@ int get_frame_drop_info(const char *infile, int *num_video_frames,
...
   *normalized_frame_drop_average_length = 0.0;
+  *frame_drop_length_percentile = 0.0;
   if (drop_length_sec_list.size() > 0) {
     float frame_drop_average_length =
         drop_length_sec_list_sum / drop_length_sec_list.size();
     *normalized_frame_drop_average_length =
         (frame_drop_average_length / delta_timestamp_sec_median);
+  
+    sort(drop_length_sec_list.begin(), drop_length_sec_list.end());
+    int position = (percentile / 100.0) * drop_length_sec_list.size();
```
Comment (chemag) [inline src/liblcvm.cc:231]: please move these 3 lines to a separate function

**Review by chemag — COMMENTED** [review]
mostly ok, with some nits

File: src/liblcvm.cc (hunk 3 above)
Comment (chemag) [reply src/liblcvm.cc:217]: What about?

    *frame_drop_count = drop_length_sec_list.size()

This may be +1 or -1. Please make sure you test it

Comment (leela9980) [reply src/liblcvm.cc:217]: drop_length_sec_list this has only cluster values, i thought requirement was to get exact number frames dropped.
ex: if there are 2 cluster frame drops but could have 10 frames dropped.

Comment (chemag) [reply src/liblcvm.cc:217]: delta_timestamp_sec_list.size()

?

Comment (leela9980) [reply src/liblcvm.cc:217]: No, ```delta_timestamp_sec_list``` contains  interframe difference of all the frames not just frame drops or clusters.
this is how we calculate total duration

```python
 float total_duration_sec = 0.0;
  for (const auto &delta_timestamp_sec : delta_timestamp_sec_list) {
    total_duration_sec += delta_timestamp_sec;
  }
```

**Review by chemag — APPROVED** [review]

**Pull request merged**